### ISSUE TITLE
Names for cori machine files

### ISSUE THREAD

**Issue by ndkeen** [E3SM-Project/scream#1508]
I see a `components/scream/cmake/machine-files/cori.cmake`, but I found I needed
a `components/scream/cmake/machine-files/cori-knl.cmake` to build on cori-knl (just a copy).

**Comment by bartgol** [comment]
That's odd. I thought we were currently testing on cori-knl on a nightly basis. Perhaps @jgfouca can comment.

**Comment by jgfouca** [comment]
@ndkeen , @bartgol , the new cmake_macros system is hierarchical and each level in the hierarchy is optional. Roughly, the levels are:
1) universal.cmake
2) $compiler.cmake
3) $machine.cmake
4) $compiler_$machine.cmake

Historically, we (E3SM) have put most of the settings in the `$compiler_$machine.cmake` files (previously, this was the machine+compiler block of XML in config_compilers.xml) and the machine-wide settings ($machine.cmake) we usually skipped or had very little in them.

Looking at our files, we do have a `intel_cori-knl.cmake` file which should allow our nightlies to work. That said, @ndkeen , if there's a setting in `cori.cmake` that you think is needed for all cori-knl builds/compilers, we should make the copy as you have done. It might be better to just rename the file since `cori` is not even a valid machine. Looking at E3SM master, the `cori.cmake` does not exist, so this was likely a mistake made on the SCREAM side of things. I will fix it.

**Comment by ndkeen** [comment]
The file in question here is `components/scream/cmake/machine-files/cori.cmake`

When I try `create_test SMS_D_P12x8_Ln4.ne30_ne30.F2000SCREAMv1`
I will get build error with:

```

Found scream component
CMake Error at cmake/build_scream.cmake:10 (include):
  include could not find requested file:

    /global/cscratch1/sd/ndk/wacmy/s48-mar24/components/scream/cmake/machine-files/cori-knl.cmake
Call Stack (most recent call first):
  CMakeLists.txt:107 (build_scream)

```

**Comment by bartgol** [comment]
I think @jgfouca fixed this with #1513.
